src/main/kotlin/com/cognifide/gradle/aem/instance/InstanceSync.kt

```diff
@@ -15,7 +14,21 @@ import org.zeroturnaround.zip.ZipUtil
 @Suppress("LargeClass", "TooManyFunctions")
 class InstanceSync(aem: AemExtension, instance: Instance) : InstanceHttpClient(aem, instance) {
 
-    fun determineRemotePackage(file: File, refresh: Boolean = true): Package? {
+    fun getPackage(file: File, refresh: Boolean = true, retry: Retry = aem.retry()): Package {
```

Comment:
Consider grouping refresh and retry in one object, e.g. exaecParams/policy. They do now have a business meaning, they are more technical detail.

Reply:
yep, agree, I will consider refactoring it in future. thx

Reply:
grouping it into `ListPackageOptions.()` lambdas 

`UploadOptions.()` etc